controllers/storagecluster/cephcluster.go

```diff
@@ -994,7 +995,8 @@ func newStorageClassDeviceSets(sc *ocsv1.StorageCluster) []rookCephv1.StorageCla
 			} else {
 				ds.Count = sc.Spec.ManagedResources.CephNonResilientPools.Count
 			}
-			if sc.Spec.ManagedResources.CephNonResilientPools.Resources != nil {
+			if sc.Spec.ManagedResources.CephNonResilientPools.Resources != nil &&
```

Comment:
Does it work,

1) 4.18 Replica one was enabled and upgraded
2) 4.18 Replica one was disabled and upgraded
3) 4.19 Replica one is enabled
4) 4.19 Replica one is disabled

Be extra cautious to all the test on a running cluster before merging